src/control/lib/hardware/pciutils/bindings.go

```diff
@@ -0,0 +1,120 @@
+//
+// (C) Copyright 2024 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package pciutils
+
+import (
+	"os"
+	"unsafe"
+
+	"github.com/daos-stack/daos/src/control/lib/hardware"
+	"github.com/pkg/errors"
+)
+
+/*
+#cgo LDFLAGS: -lpci
+
+#include <stdlib.h>
+#include <pci/pci.h>
+*/
+import "C"
+
+type api struct{}
+
+func (api *api) Cleanup() {}
+
+func speedToFloat(speed uint16) float32 {
+	var mant float32
+
+	switch speed {
+	case 1:
+		mant = 2.5
+	case 2:
+		mant = 5
+	case 3:
+		mant = 8
+	case 4:
+		mant = 16
+	case 5:
+		mant = 32
+	case 6:
+		mant = 64
+	default:
+		return 0
+	}
+
+	return mant * 1e+9
+}
+
+func (api *api) PCIeCapsFromConfig(cfgBytes []byte, dev *hardware.PCIDevice) error {
```

Comment:
Don't forget to add doc comments to any new exported functions

Reply:
fixed